services/src/main/java/org/keycloak/protocol/oidc/endpoints/request/AuthzEndpointRequestParser.java

```diff
@@ -120,23 +184,61 @@ protected void validateResponseTypeParameter(String responseTypeParameter, Autho
     }
 
     protected void extractAdditionalReqParams(Map<String, String> additionalReqParams) {
+        int currentAdditionalReqParamMaxOverallSize = 0;
         for (String paramName : keySet()) {
-            if (!KNOWN_REQ_PARAMS.contains(paramName)) {
-                String value = getParameter(paramName);
-                if (value != null && value.trim().isEmpty()) {
-                    value = null;
-                }
-                if (value != null && value.length() <= ADDITIONAL_REQ_PARAMS_MAX_SIZE) {
-                    if (additionalReqParams.size() >= ADDITIONAL_REQ_PARAMS_MAX_MUMBER) {
-                        logger.debug("Maximal number of additional OIDC params (" + ADDITIONAL_REQ_PARAMS_MAX_MUMBER + ") exceeded, ignoring rest of them!");
-                        break;
-                    }
-                    additionalReqParams.put(paramName, value);
-                } else {
-                    logger.debug("OIDC Additional param " + paramName + " ignored because value is empty or longer than " + ADDITIONAL_REQ_PARAMS_MAX_SIZE);
-                }
+          
+          if (KNOWN_REQ_PARAMS.contains(paramName)) {
+            logger.debugv("The additional OIDC param ''{0}'' is well known. Continue with the other additional parameters.", paramName);
+            continue;
+          }
+          
+          final String value = getParameter(paramName);
+          
+          if (value == null || value.trim().isEmpty()) {
+            logger.debugv("The additional OIDC param ''{0}'' ignored because it's value is null or blank.", paramName);
+            continue;
+          }
+
+          // Compare with ">=", as the currently processed parameter will be added at the END of this method.
+          if (additionalReqParams.size() >= additionalReqParamsMaxNumber) {
+            
+            if (additionalReqParamsFailFast) {
+              logger.infov("The maximum number of allowed parameters ({0}) is exceeded.", additionalReqParamsMaxNumber);
```

Comment:
Is it possible to use `logger.warnv` instead? Usually `info` is supposed to be used for server-wide important information (EG. `Keycloak server successfully started in X seconds`) rather than error-like scenarios.

Similar likely applies to other `info` messages introduced in this PR.

Reply:
As the client sent too many/too long request parameters, the KC client made a mistake, not the KC server. I think the KC server log should only contain an INFO that the client made a mistake, not a WARN. I would recommend to log to INFO rather to WARN.
A WARN log is always something the (keycloak) server owner should have a look at, as something may not be right on the server. That is certainly not the case here.

E.g.,
If the KC logfile is being monitored for WARN/ERROR logs, the monitor would alarm everytime a client does such a mistake. If the client contacts the keycloak owner about an error, the keycloak owner is still able to help/respond by checking the logs for the particilar keycloak client/timestamp.